File: app/src/main/java/com/steevsapps/idledaddy/listeners/AndroidLogListener.java
package com.steevsapps.idledaddy.listeners;

import android.util.Log;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import in.dragonbra.javasteam.util.log.LogListener;

/**
 * LogListener that prints to Android Logcat
 */
public class AndroidLogListener implements LogListener {
    private final static String TAG = "JavaSteam";

    @Override
    public void onLog(@NonNull Class<?> clazz, @Nullable String message, @Nullable Throwable throwable) {
        String threadName = Thread.currentThread().getName();
        threadName = threadName.substring(0, Math.min(10, threadName.length()));
        String className = clazz.getName();

        if (message == null) {
            Log.i(TAG, String.format("[%10s] %s", threadName, className), throwable);
        } else {
            Log.i(TAG, String.format("[%10s] %s - %s", threadName, className, message), throwable);
        }
    }

    @Override
    public void onError(@NonNull Class<?> clazz, @Nullable String message, @Nullable Throwable throwable) {
        String threadName = Thread.currentThread().getName();
        threadName = threadName.substring(0, Math.min(10, threadName.length()));
        String className = clazz.getName();

        if (message == null) {
            Log.e(TAG, String.format("[%10s] %s", threadName, className), throwable);
        } else {
            Log.e(TAG, String.format("[%10s] %s - %s", threadName, className, message), throwable);
        }
    }
}

File: app/src/main/java/com/steevsapps/idledaddy/listeners/AndroidLogListener.kt
package com.steevsapps.idledaddy.listeners

import android.util.Log
import `in`.dragonbra.javasteam.util.log.LogListener

/**
 * LogListener that prints to Android Logcat
 */
class AndroidLogListener : LogListener {
    override fun onLog(clazz: Class<*>, message: String?, throwable: Throwable?) =
        log(clazz, message, throwable, Log::i)

    override fun onError(clazz: Class<*>, message: String?, throwable: Throwable?) =
        log(clazz, message, throwable, Log::e)

    private fun log(
        clazz: Class<*>,
        message: String?,
        throwable: Throwable?,
        logType: (String, String, Throwable?) -> Int
    ) {
        val threadName = Thread.currentThread().name.take(10)
        val className = clazz.name
        val logMessage = if (message == null) {
            "[%10s] %s".format(threadName, className)
        } else {
            "[%10s] %s - %s".format(threadName, className, message)
        }
        logType(TAG, logMessage, throwable)
    }

    companion object {
        private const val TAG = "JavaSteam"
    }
}